src/main/scala/org/constellation/primitives/RandomTransactionManager.scala

```diff
@@ -211,4 +211,146 @@ class RandomTransactionManager[T](nodeActor: ActorRef, periodSeconds: Int = 10)(
       "randomTransactionLoop"
     )
   }
+
+  def generate(): Unit =
+    for {
+      _ <- EitherT.fromEither(validateNodeState(NodeState.Ready))
+      _ <- EitherT.fromEither(validateNodeIsPermitToGenerateRandomTransaction)
+
+      readyPeers <- EitherT.liftF(getReadyPeers)
+      _ <- validateNodeIsNonEmptyOrIsGenesisNode(readyPeers)
+
+      address <- EitherT.liftF(getAddressNode)
+      _ <- validateNodeHasBalance(address)
+
+      pendingTransactionsCount <- EitherT.liftF(getCountTransactionsWithStatus(TransactionStatus.Pending))
+      _ <- validatePendingNumberOfTransactionIsLessThanMemPool(pendingTransactionsCount)
+
+//      transactions <- createTransactions(readyPeers)
+    } yield ()
+
+//  private def createTransactions(peerIds: Seq[(Id, PeerData)]): IO[Transaction] =
+//    if (multiAddressGenerationMode) {
+//      for {
+//        balancesAddresses <- balancesForAddresses
+//      } yield createMultipleAddressTransaction(balancesAddresses, addressesHaveSufficient(balancesAddresses), peerIds)
+//    } else {
+//      IO.pure(createSingleAddressTransaction(dao.selfAddressStr, createRandomAddressFromPeers(peerIds)))
+//    }
+
+  private def balancesForAddresses: IO[Seq[(String, Option[AddressCacheData])]] =
```

Comment:
It's better to rely directly on `addressService: AddressService[F]` rather than on `dao.addressService`. This way you'll have `F[...]` instead of `IO[...]` here so you don't need to convert it later using `LiftIO`